jetty-core/jetty-deploy/src/main/java/org/eclipse/jetty/deploy/providers/ContextProvider.java

```diff
@@ -641,4 +580,308 @@ protected void pathChanged(Path path) throws Exception
         if (isDeployable(path))
             super.pathChanged(path);
     }
+
```

Comment:
For the `isDeployable` method, as you're doing a lot of rewriting, it may be worth looking into #12543. The problem here is that on a hot redeploy via touching just the war (rather than the associated xml file),  line 562 will cause the redeploy not to happen. That line works fine on initial deployment, but not on a redeploy.